google/cloud/aiplatform/metadata/metadata.py

```diff
@@ -304,6 +385,73 @@ def _get_experiment_or_pipeline_resource_name(
             )
         return context.resource_name
 
+    def _query_runs_to_data_frame_v2(
+        self, context_id: str, context_resource_name: str, source: str
+    ) -> "pd.DataFrame":  # noqa: F821
+        """Get metrics and parameters associated with a given Context into a Dataframe.
+
+        Compatible with Experiment v2 data model.
+
+        Args:
+            context_id (str):
+                Name of the Experiment or Pipeline.
+            context_resource_name (str):
+                Full resource name of the Context associated with an Experiment or Pipeline.
+            source (str):
+                Identify whether the this is an Experiment or a Pipeline.
+
+        Returns:
+            The full resource name of the Experiment or Pipeline Context.
+
+        Raises:
+            ImportError: If pandas is not installed.
+        """
+        try:
+            import pandas as pd
+        except ImportError:
+            raise ImportError(
+                "Pandas is not installed and is required to get dataframe as the return format. "
+                'Please install the SDK using "pip install python-aiplatform[metadata]"'
+            )
+
+        filter = f'schema_title="{constants.SYSTEM_EXPERIMENT_RUN}" AND parent_contexts:"{context_resource_name}"'
+        run_contexts = _Context.list(filter=filter)
+
+        in_context_query = " OR ".join(
+            [f'in_context("{c.resource_name}")' for c in run_contexts]
+        )
+
+        filter = f'schema_title="{constants.SYSTEM_RUN}" AND ({in_context_query})'
+
+        run_executions = _Execution.list(filter=filter)
+
+        context_map = {c.name: c for c in run_contexts}
+
+        context_summary = []
+        for run_execution in run_executions:
+            run_context = context_map[run_execution.name]
```

Comment:
Another possible way I was thinking here is that we could just loop through the run contexts directly, and list run executions individually during each loop, this way, we do not need to link run context and run executions through its display name. But it might have some latency hit as we are making more API calls.